### DISCUSSION TITLE
error: RPC failed; curl 18 transfer closed with outstanding read data remaining

### DISCUSSION THREAD

**Discussion by simonnchong** [community/community#28641] [Other Feature Feedback, Questions, & Ideas] [answered]
I got this error when I wanna push a new repository to my empty origin. May I know what is the problem? I'd tried to increase the buffer size but still, nothing solved.

$ git push -u origin main
Uploading LFS objects: 100% (311/311), 37 MB | 0 B/s, done.
Enumerating objects: 13272, done.
Counting objects: 100% (13272/13272), done.
Delta compression using up to 16 threads
Compressing objects: 100% (7048/7048), done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
send-pack: unexpected disconnect while reading sideband packet
Writing objects: 100% (13272/13272), 2.19 GiB | 31.45 MiB/s, done.
Total 13272 (delta 6196), reused 13272 (delta 6196), pack-reused 0
fatal: the remote end hung up unexpectedly
Everything up-to-date

**Comment by airtower-luna** [comment, marked as answer]
>  Writing objects: 100% (13272/13272), 2.19 GiB | 31.45 MiB/s, done.

This looks like your push is significantly larger than [the limit imposed by GitHub](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#file-size-limits):

> GitHub blocks pushes that exceed 100 MB.

If that's because of a long history with lots of small files, you might be able to push some older commits until you get to the current one, so each push is smaller. If you have files that are too large you'll either have to move them to Git LFS and remove them from the history (also described in the linked documentation), or find some other hosting.

**Reply by simonnchong** [reply]
seems I can't do it anyway, thanks for your replying!
